### DISCUSSION TITLE
(Arch, KDE) launched program does not match pinned shortcut

### DISCUSSION THREAD

**Discussion by velamyGH** [ppy/osu#28397] [Q&A] [answered]
### Type

Other

### Bug description

when osu!lazer is pinned to the KDE default taskbar (Icons-only Task Manager), the launched program runs under a separate icon which for some reason is using the old logo as well, not sure what this would be pulling from but i have a suspicion its from stable under wine (this specifically if that helps https://github.com/NelloKudo/osu-winello), although this is still unintended behavior regardless.

- KDE ver: Plasma 6.0.5 (although this has always happened with lazer, at least since plasma 5)

- CPU: AMD Ryzen 5 3600, GPU: AMD Radeon RX 6500XT (just in case)

### Screenshots or videos

![image](https://github.com/ppy/osu/assets/48131627/a15f30d1-1a8b-4911-9413-24459ffe0b6a)

### Version

2024.521.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/15572870/compressed-logs.zip)

**Comment by umuod** [comment]
I was able to reproduce this by installing osu-wine from the mentioned repos. ~~Seems to be fixed by changing osu-wine's osu! directory to not be the same as osu!lazer (in my case it was `~/.local/share/`). Try and see if that helps with your issue.~~

**Comment by velamyGH** [comment]
> I was able to reproduce this by installing osu-wine from the mentioned repos. Seems to be fixed by changing osu-wine's osu! directory to not be the same as osu!lazer (in my case it was `~/.local/share/`). Try and see if that helps with your issue.

tried moving the directories, didnt seem to fix the issue. here are my directories in case i am misunderstanding though
![image](https://github.com/ppy/osu/assets/48131627/3675090d-26d3-4c79-aec5-eedd548ebab6)
![image](https://github.com/ppy/osu/assets/48131627/9a1633d6-8ff8-46c4-97a0-2e86dd2440f2)

**Comment by umuod** [comment]
I made a mistake. ~~I think the real issue was that osu was taking the icon from `~/.local/share/icons/` as renaming a random image as `osu-wine.png` there cause osu!lazer shortcut to launch with that icon. I think maybe the hacky way of putting osu!lazer icon there might work. It seems to be for me.~~

**Comment by velamyGH** [comment]
adding a . at the end of osu-wine.png does make the app launch with the intended icon, however the shortcut still creates a second program on the taskbar, and pinning the launched app makes the shortcut no longer work.

![image](https://github.com/ppy/osu/assets/48131627/e33e8e0b-5213-4091-b4c5-e9d7a4140dd1)

**Comment by umuod** [comment]
You are right. It seems to be a 50/50 for me whether or not lazer launch matches the pinned icon or not so I apologize for the inconsistent suggestions. Though I noticed both my lazer .desktop file and osu-wine.desktop file had the same `Name=osu!`. I changed osu-wine's and after opening the shortcut about twenty times osu!lazer seems to be matching with the pinned icon without creating a seperate icon so maybe that was the issue? Could you try this too?

**Comment by velamyGH** [comment]
> You are right. It seems to be a 50/50 for me whether or not lazer launch matches the pinned icon or not so I apologize for the inconsistent suggestions. Though I noticed both my lazer .desktop file and osu-wine.desktop file had the same `Name=osu!`. I changed osu-wine's and after opening the shortcut about twenty times osu!lazer seems to be matching with the pinned icon without creating a seperate icon so maybe that was the issue? Could you try this too?

no worries about the suggestions, i appreciate it either way!! best to try as much as possible just in case :) as for conflicting shortcut information i cant seem to find anything that would cause that.. heres the info if you wanna take a look
![image](https://github.com/ppy/osu/assets/48131627/6432ed15-ce75-4632-8656-e7aa1a7afa6b)
![image](https://github.com/ppy/osu/assets/48131627/cbc34419-c0d3-4425-a141-20656b53a501)
![image](https://github.com/ppy/osu/assets/48131627/e9358b44-2791-4c69-a5b3-99cdb494c27d)

**Comment by umuod** [comment]
The issue happens if I use your name combination. Though it seems to be fine when I tried a dozen more times with this name combination. I also launch through the appimage as opposed to your arch-packaged it seems. 
![Screenshot_20240605_152813](https://github.com/ppy/osu/assets/161921855/042377ce-4762-4096-8994-6c267c2827b5)

**Reply by velamyGH** [reply]
> The issue happens if I use your name combination. Though it seems to be fine when I tried a dozen more times with this name combination. I also launch through the appimage as opposed to your arch-packaged it seems. ![Screenshot_20240605_152813](https://private-user-images.githubusercontent.com/161921855/336712808-042377ce-4762-4096-8994-6c267c2827b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc1Njk1MDEsIm5iZiI6MTcxNzU2OTIwMSwicGF0aCI6Ii8xNjE5MjE4NTUvMzM2NzEyODA4LTA0MjM3N2NlLTQ3NjItNDA5Ni04OTk0LTZjMjY3YzI4MjdiNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwNVQwNjMzMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jY2Y2ZjFlMmVlNTc1NzJhYzRkZTlhZWY1ZGYyOGE3MGZmNjNiMjJmNjY3ZTk3YTk1M2VlNzYyNDM1MjhhODNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lNetPv17f3KVkmaV77ST3k4VL1k_NRAibuFUxqPqg2A)

trying this doesnt seem to work for me, i think this also happened when i used the appimage before switching to the aur release?

![image](https://github.com/ppy/osu/assets/48131627/2dcda384-d6e5-4aaa-8a1b-2eed89850125)

**Comment by umuod** [comment, marked as answer]
can you try adding `StartupWMClass=osu!` in the osu!lazer shortcut .desktop? Seems to be working from another dozen attempts I tried with your name combination. From this [source](https://beamtic.com/duplicated-icons-in-launcher)

**Reply by velamyGH** [reply]
> can you try adding `StartupWMClass=osu!` in the osu!lazer shortcut .desktop? Seems to be working from another dozen attempts I tried with your name combination. From this [source](https://beamtic.com/duplicated-icons-in-launcher)

yup doing this worked! i had to make sure it was the shortcut in `~/.local/share/applications` though. thank you!
